src/main/java/org/folio/rest/impl/PiecesAPI.java

```diff
@@ -91,9 +99,21 @@ public void putOrdersPiecesById(String pieceId, boolean createItem, boolean dele
   @Override
   @Validate
   public void deleteOrdersPiecesById(String pieceId, boolean deleteHolding, Map<String, String> okapiHeaders,
-      Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+                                             Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
     pieceDeleteFlowManager.deletePiece(pieceId, deleteHolding, new RequestContext(vertxContext, okapiHeaders))
       .onSuccess(ok -> asyncResultHandler.handle(succeededFuture(buildNoContentResponse())))
       .onFailure(fail -> handleErrorResponse(asyncResultHandler, fail));
+
+  }
+
+  @Override
+  @Validate
+  public void deleteOrdersPiecesBatch(PieceCollection entity, Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    List<String> pieceIds = new ArrayList<>();
+    for (Piece piece : entity.getPieces()) {
+      pieceIds.add(piece.getId());
+    }
```

Comment:
I believe we can do by using steam instead of for loop. that will be consice

Reply:
good catch! will do

Reply:
good catch, will do!